src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -29,13 +29,17 @@ class ProductRepository extends BaseProductRepository implements ProductReposito
      */
     public function createByTaxonPaginator(TaxonInterface $taxon, array $criteria = [])
     {
+        $root = $taxon->isRoot() ? $taxon : $taxon->getRoot();
+
         $queryBuilder = $this->createQueryBuilder('o');
         $queryBuilder
             ->innerJoin('o.taxons', 'taxon')
+            ->andWhere($queryBuilder->expr()->eq('o.root', ':root'))
```

Comment:
[TaxonRepository](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php#L32)

Isn't quite the same thing?